specs/Slots.md

```diff
@@ -0,0 +1,64 @@
+# Slots API specification
+
+Fluent UI components almost always contain sub parts and these pre-defined layouts can be configured via Slots API. Slots are named areas in a component that can receive props and provide styling and layout for them.
+
+## Slots definition
+
+For example, `Button` component contains `content`, `loader` and `icon` slots in its layout:
+
+```jsx
+<root>
+  <loader />
+  <content />
+  <icon />
+</root>
+```
+
+## Slots usage
+
+Slots can be configured through props, which lets the caller pass in a variety of inputs for a given slot. There are several forms of values that can be provided, but all of them share one common thing - they will produce a proper JSX layout inside a component.
+
+### An object as a value
+
+If you will pass an object to component props we will handle it as props for a slot. In an example below, props will be passed to an `icon` slot:
+
+```jsx
+<Button icon={{ children: <FooIcon />, id: '#button-icon' }} />
+```
+
+### Primitives as a value
+
+We consider JSX elements, strings and numbers as primitive values that will be passed to a meaningful prop, usually to `children`:
+
+```jsx
+<>
+  <Button icon={<FooIcon />} />
+  {/* 💡 has identical effect to the previous one */}
+  <Button icon={{ children: <FooIcon /> }} />
+</>
+```
+
+To disable slot rendering you can use falsy values:
```

Comment:
**q:** falsy or null ? I'd recommend to explicit/follow js language practices -> thus allow `null` only

Reply:
I modified this section to be more explicit: "To disable slot rendering you can use falsy (`null`, `false`) values:" and also mentioned `true` behavior below.

We are already handling `false` & `null` in this way I think that we should include `false` as this behavior can be unclear for customers:

```jsx
<>
  <Button loader={null} /> // hides slot
  <Button loader={false} /> // will not hide a slot if will not include "false"
</>
```

Reply:
I resolved a separate comment referring to this issue after seeing this one again.

@layershifter Does this mean that only `null, false` are supported ? In this case I would refrain from using the term `falsy` since it relates to a larger set of possible values (0, empty string)